tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyTool.java

```diff
@@ -13,12 +13,49 @@
 import javax.servlet.http.HttpServletResponse;
 
 import org.json.JSONException;
+import org.unicode.cldr.util.CLDRConfig;
 import org.unicode.cldr.util.CldrUtility;
 import org.unicode.cldr.util.VettingViewer;
 
 public class SurveyTool extends HttpServlet {
     private static final long serialVersionUID = 1L;
-    public static final boolean USE_DOJO = true;
+    private static final boolean USE_DOJO_DEFAULT = true;
+
+    /**
+     * do we use Dojo? (lacking any other context)
+     * Better to use the other variants taking a WebContext or an HttpServletRequest
+     * @deprecated this will be going away
+     * @return
+     */
+    @Deprecated
+    public static final boolean useDojo() {
+        return CLDRConfig.getInstance().getProperty("USE_DOJO", USE_DOJO_DEFAULT);
+    }
+
+    /**
+     * do we use Dojo?
+     * @param ctx request context
+     * @deprecated this will be going away
+     * @return
+     */
+    @Deprecated
+    public static final boolean useDojo(WebContext ctx) {
+        return useDojo(ctx.request);
+    }
+    /**
+     * do we use Dojo?
+     * @param ctx request context
+     * @deprecated this will be going away
+     * @return
+     */
+    @Deprecated
+    public static final boolean useDojo(HttpServletRequest ctx) {
+        String param = ctx.getParameter("USE_DOJO");
```

Comment:
I'm guessing there's an easy way for the user to add "&USE_DOJO=false" to all the urls? I mean not just in the address bar, but in all the automatic xhr requests, etc? How is that done?

Reply:
shouldn't affect XHRs? or does it.

Reply:
Yes, it can affect XHRs, such as WHAT_ADMIN_PANEL. I think if USE_DOJO is false, then WHAT_ADMIN_PANEL is going to behave as though USE_DOJO were true, and the front and back ends won't cooperate.

A solution might be to add "&USE_DOJO=false" to all urls in cldrAjax.js, only for the non-dojo version of cldrAjax.js. Currently we use the same cldrAjax.js for dojo and non-dojo. We could clone it as CldrDojoAjax.js, or add a USE_DOJO constant to the front end...

Reply:
SurveyAjax.java has:
```
                    if (what.equals(WHAT_ADMIN_PANEL)) {
...
                            if (SurveyTool.USE_DOJO) {
                                response.sendRedirect(request.getContextPath() + "/AdminPanel.jsp" + "?vap=" + SurveyMain.vap);
                            } else {
...
                                JSONWriter r = newJSONStatus(request, sm);
...
                                new AdminPanel().getJson(r, request, response, sm);
```

the useDojo(request) method can't be used in place of USE_DOJO here unless the request specifies USE_DOJO.

If we really want to make the dojo/non-dojo switch a request param, we'd better make a clean break between dojo and non-dojo versions of all the js files; sharing would be too error-prone. Only these 5 are currently shared:

* new/cldrAjax.js
* new/cldrCsvFromTable.js
* new/cldrForumFilter.js
* new/cldrStatus.js
* new/cldrText.js

Maybe generalize the request syntax: `..&frontend=dojo` or `..&frontend=vue`or `..&frontend=default`